scripts/fonts.js

```diff
@@ -0,0 +1,143 @@
+          );
+          return;
+        }
+        console.log(`stdout: ${stdout}`);
+      }
+    );
+    console.log(`Downloaded ${fontFamily}`);
+    fs.writeFileSync(downloadLockFile, "");
+  }
+}
+
+const downloadLockFolder = "download/font-lock";
+const downloadFolder = "download/font";
+const distFontFolder = "dist/fonts";
+const ttfFontFolder = "build/font/ttf";
+
+[downloadFolder, ttfFontFolder, distFontFolder, downloadLockFolder].forEach(
+  (folder) => {
+    fs.mkdirSync(folder, { recursive: true }, (e) => {});
+  }
+);
+
+const fontDef = JSON.parse(fs.readFileSync("scripts/fonts.json"));
+const fontRanges = fontDef["glyph-ranges"];
+const customFontStacks = fontDef["custom-font-stacks"];
+const bundleFontStacks = fontDef["bundle-font-stacks"];
+const fontFamilies = fontDef["font-families"];
+
+loadGoogleFonts(fontFamilies, downloadFolder);
+
+//Just copy fonts that are used unmodified
+loadGoogleFonts(bundleFontStacks, ttfFontFolder);
+
+for (const stack in customFontStacks) {
+  let font;
+  for (const stackPart in customFontStacks[stack]) {
+    let stackPartDef = customFontStacks[stack][stackPart];
+    const inputFontBuffer = fs.readFileSync(
+      `${downloadFolder}/${stackPartDef.file}.ttf`
+    );
+
+    const ranges = stackPartDef.ranges;
+    for (const range in ranges) {
+      const namedRange = ranges[range];
+      const numberedRanges = fontRanges[namedRange];
+      for (const numberedRange in numberedRanges) {
+        const subsetRange = numberedRanges[numberedRange];
+        const subsetGlyphArray = createRangeArray(
+          subsetRange[0],
+          subsetRange[1]
+        );
+        const fontSegment = Font.create(inputFontBuffer, {
+          type: "ttf",
+          subset: subsetGlyphArray,
+          hinting: true,
+          compound2simple: true,
+          inflate: null,
+          combinePath: false,
+        });
+        if (font === undefined) {
+          font = fontSegment;
+          font.data.name = {
+            copyright: "Repackaged from Google Fonts",
+            fontFamily: "Americana",
+            fullName: stack,
+            manufacturer: "OpenStreetMap Americana Team",
+          };
+        } else {
+          font.merge(fontSegment);
+        }
+      }
+    }
+  }
+
+  // write font file
+  const outputBuffer = font.write({
+    // support ttf, woff, woff2, eot, svg
+    type: "ttf",
+    // save font hinting
+    hinting: true,
+    // deflate function for woff, eg. pako.deflate
+    deflate: null,
+    // for user to overwrite head.xMin, head.xMax, head.yMin, head.yMax, hhea etc.
+    support: { head: {}, hhea: {} },
+  });
+
+  const ttfFile = `build/font/ttf/${stack}.ttf`;
+  fs.writeFileSync(ttfFile, outputBuffer);
+  console.log(`Built ${ttfFile}`);
+}
+
+//Binary download.  Temporary hack until https://github.com/stadiamaps/build_pbf_glyphs/issues/4 is resolved
+const pbfBuilderFilename = "~/.cargo/bin/build_pbf_glyphs";
+
+execSync(
```

Comment:
Not well documented, but the Promise version of `exec` allows removing all of this boilerplate.

https://nodejs.org/dist/latest-v19.x/docs/api/child_process.html#child_processexeccommand-options-callback

Reply:
I re-wrote this in e1073d9.

Okay, actually in the interest of full disclosure, I didn't actually understand your comment so I fed it to ChatGPT and it rewrote it for me.  Hope we got it right...

Reply:
Lol, lmao.

I would do `const exec = promisify(child_process.exec)` rather than at every call site.